index.js

```diff
@@ -6,25 +6,25 @@ module.exports = {
   /**
    * Get information about the OpenSSL version installed where this module is run.
    * Returns a promise.
-   * @returns {promise}
+   * @returns {Promise<{ ciphers: { supported: string[], unsupported: string[] }, protocols: { supported: string[], unsupported: string[] }, version: string }>}
    */
   getOpenSSLCapabilities: function() {
     return opensslinfo.getOpenSSLCapabilities();
   },
   /**
    * Determine the supported SSL/TLS protocols and ciphers for a server.
    * Returns a promise.
-   * @param hostData {{ host: string, port: number, [servername]: string }}
-   * @returns {promise}
+   * @param hostData {{ host: string, port: number, servername?: string, rejectUnauthorized?: boolean, minDHSize?: number }}
```

Comment:
Neither `rejectUnauthorized` nor `minDHSize` are parameters that have been documented as being supported. Technically, any option that can be passed to `tls.connect()` can be passed here, but the goal is not to be exhaustive about that. The documented options are the ones officially supported, and something can be added to the README pointing out that extra options can be supplied.

Reply:
K, but i can't get intellisense for that options because the JSDoc explicitly states that the parameter `hostData` expects an object of `{ host: string, port: number, [servername]: string }`.

If a dev is using a linter like TypeScript or ESlint he/she will be getting an error for setting a property like `minDHSize` not because it's not supported but because the JSDoc does not define this property.